File: backend/app/schemas/analytics.py
import uuid
from datetime import date

from pydantic import BaseModel


class BurndownPoint(BaseModel):
    date: date
    remaining_points: int


class SprintBurndown(BaseModel):
    sprint_id: uuid.UUID
    sprint_name: str
    status: str
    total_items: int
    completed_items: int
    completion_rate: float | None
    capacity_points: int | None
    total_points: int
    completed_points: int
    burndown_series: list[BurndownPoint] = []


class BottleneckItem(BaseModel):
    id: uuid.UUID
    title: str
    status: str
    days_in_status: float
    sprint_id: uuid.UUID | None


class ProjectAnalytics(BaseModel):
    active_sprint: SprintBurndown | None
    bottlenecks: list[BottleneckItem]
    velocity: float | None

File: frontend/src/api/analytics.ts
import { apiClient } from "./client";

export interface BurndownPoint {
  date: string;
  remaining_points: number;
}

export interface SprintBurndown {
  sprint_id: string;
  sprint_name: string;
  status: string;
  total_items: number;
  completed_items: number;
  completion_rate: number | null;
  capacity_points: number | null;
  total_points: number;
  completed_points: number;
  burndown_series: BurndownPoint[];
}

export interface BottleneckItem {
  id: string;
  title: string;
  status: string;
  days_in_status: number;
  sprint_id: string | null;
}

export interface ProjectAnalytics {
  active_sprint: SprintBurndown | null;
  bottlenecks: BottleneckItem[];
  velocity: number | null;
}

export async function getProjectAnalytics(projectId: string): Promise<ProjectAnalytics> {
  const { data } = await apiClient.get<ProjectAnalytics>(`/projects/${projectId}/analytics`);
  return data;
}
